frontends/android/BitBoxApp/app/src/main/java/ch/shiftcrypto/bitboxapp/MainActivity.java

```diff
@@ -358,12 +358,39 @@ public void onChanged(Boolean isDarkTheme) {
             }
         });
 
+        goViewModel.getRequestAuth().observe(this, new Observer<Boolean>() {
+            @Override
+            public void onChanged(Boolean requestAuth) {
+                Util.log("requested Auth: " + requestAuth);
+                if (requestAuth) {
+                    goViewModel.setRequestAuth(false);
```

Comment:
I don't like much using a MutableLiveData to trigger the requestAuth, but I could not find a nicer way to do it.

Reply:
i guess the reason is so you can do the stuff here in MainActivity to have access to `MainActivity.this`? or is there a different reason?

this line specifically to reset the requestAuth var so the next one triggers this callback again does look a bit fishy :joy: 

i had a chat with ChatGPT: https://chat.openai.com/share/f7b28f24-d71f-4502-9fa9-0b79b0e7cc20

it commented on the fact that a device rotation etc. will trigger a onStart() again and observe here again, which means it could be called multiple times for each auth-request? did you test what happens after you rotate the screen multiple times:

- what happens to an active auth request?
- what happens to a auth request triggered after some screen rotations?

Reply:
sholdn't you set this back to `false` after auth has failed or succeeded, so that if multiple auth requests happen in parallel, only one is processed at once?

Reply:
> i guess the reason is so you can do the stuff here in MainActivity to have access to MainActivity.this? or is there a different reason?

Yes, exactly.

I tested device rotations, but those do not seem to trigger a new onStart. I never experienced issues in that sense. Anyway, I spent a lot of time trying to improve this part to make it both more readable and stable. I experienced some nasty bugs, but now it seems good enough to me.